File: OPENAPI-REST-API/swagger-client/scala-akka-http-server/src/main/scala/io/swagger/server/model/WebsiteOrderPostRequest.scala
package io.swagger.server.model


/**
 * request to place a webhosting order
 *
 * @param hostname Primary FQDN for the website. Must pass valid_hostname() and the keyword blocklist.
 * @param packageId Plan id from getNewWebsite.serviceTypes[].services_id (must be buyable).
 * @param rootpass Control-panel admin password. If blank, a random password is generated server-side.
 * @param period Billing cycle in months (1 / 6 / 12 / 24 / 36).
 * @param coupon Coupon code.
 * @param serviceOfferId Promo bundle id from getNewWebsite.serviceOffers.
 * @param script Auto-installer id (0 = none).
 * @param comment Free-form note saved on the service row.
 * @param registerDomain When true and enableDomainRegistering=true from the catalog, also registers/transfers the domain through the order.
 */
case class WebsiteOrderPostRequest (
  hostname: String,
  packageId: Int,
  rootpass: Option[String],
  period: Option[Int],
  coupon: Option[String],
  serviceOfferId: Option[Int],
  script: Option[Int],
  comment: Option[String],
  registerDomain: Option[Boolean]
)

File: OPENAPI-REST-API/openapi-client/javascript-closure-angular/API/Client/WebsiteOrderPostRequest.js
goog.provide('API.Client.WebsiteOrderPostRequest');

/**
 * request to place a webhosting order
 * @record
 */
API.Client.WebsiteOrderPostRequest = function() {}

/**
 * Primary FQDN for the website. Must pass valid_hostname() and the keyword blocklist.
 * @type {!string}
 * @export
 */
API.Client.WebsiteOrderPostRequest.prototype.hostname;

/**
 * Plan id from getNewWebsite.serviceTypes[].services_id (must be buyable).
 * @type {!number}
 * @export
 */
API.Client.WebsiteOrderPostRequest.prototype.packageId;

/**
 * Control-panel admin password. If blank, a random password is generated server-side.
 * @type {!string}
 * @export
 */
API.Client.WebsiteOrderPostRequest.prototype.rootpass;

/**
 * Billing cycle in months (1 / 6 / 12 / 24 / 36).
 * @type {!number}
 * @export
 */
API.Client.WebsiteOrderPostRequest.prototype.period;

/**
 * Coupon code.
 * @type {!string}
 * @export
 */
API.Client.WebsiteOrderPostRequest.prototype.coupon;

/**
 * Promo bundle id from getNewWebsite.serviceOffers.
 * @type {!number}
 * @export
 */
API.Client.WebsiteOrderPostRequest.prototype.serviceOfferId;

/**
 * Auto-installer id (0 = none).
 * @type {!number}
 * @export
 */
API.Client.WebsiteOrderPostRequest.prototype.script;

/**
 * Free-form note saved on the service row.
 * @type {!string}
 * @export
 */
API.Client.WebsiteOrderPostRequest.prototype.comment;

/**
 * When true and enableDomainRegistering=true from the catalog, also registers/transfers the domain through the order.
 * @type {!boolean}
 * @export
 */
API.Client.WebsiteOrderPostRequest.prototype.registerDomain;

